server/plugin/api_test.go

```diff
@@ -1306,11 +1571,55 @@ func TestEditSubscription(t *testing.T) {
 					return fmt.Errorf("new error")
 				})
 			},
+			SetupPlugin:          func(p *Plugin) {},
 			ExpectedStatusCode:   http.StatusBadRequest,
 			ExpectedErrorMessage: "new error",
 		},
+		"unable to get permissions for the channel": {
+			RequestBody: fmt.Sprintf(`{
+				"user_id": "%s",
+				"channel_id": "%s"
+			  	}`, testutils.GetID(), testutils.GetChannelID()),
+			SetupAPI: func(api *plugintest.API) {},
+			SetupClient: func(client *mock_plugin.Client) {
+				var s *serializer.SubscriptionPayload
+				monkey.PatchInstanceMethod(reflect.TypeOf(s), "IsValidForUpdation", func(_ *serializer.SubscriptionPayload, _ string) error {
+					return nil
+				})
```

Comment:
Why is this inside `SetupClient`? Shouldn't it be inside `SetupPlugin`? 
Fix this everywhere.